pkg/operator/nfs/spec.go

```diff
@@ -69,6 +69,10 @@ func newServiceForNFSServer(cr *nfsv1alpha1.NFSServer) *corev1.Service {
 func newStatefulSetForNFSServer(cr *nfsv1alpha1.NFSServer) *appsv1.StatefulSet {
 	privileged := true
 	replicas := int32(cr.Spec.Replicas)
+	image := "rook/nfs:master"
```

Comment:
Instead of defaulting to master, we really need one of the following approaches:
1. Run the same image in the stateful set that is being used by the operator. 
2. Require the `image` to be set in the NFS CR.

I believe the intention is the first option. You could just detect the image in the operator with a helper that we have. Search for code calling: `k8sutil.GetContainerImage(pod, containerName)` for an example.

Or if you need option 2 with a different image from the operator , for example what image will be used?

Reply:
Thanks Travis, option 1 is perfect for my use. Do we still want to provide option 2 in case that is useful for somebody else?
I can default to option1 unless the images is provided in the CR.

Reply:
Let's just go with option 1 for now. I'm not sure the scenario for the second option.

Reply:
I am still trying to implement this, but I am having problems deciding where to do it:

- Cannot do it in spec.go, newStatefulSetForNFSServer() because I do not have the operator pod name there.
- controller.go seems to be a good option, reconcileNFSServer() , do I have the operator pod name there?
* should I blindly overwrite the image no matter what is it in the cr?

Any other ideas? Thanks

Reply:
You can access the operator pod name by env var `POD_NAME` see https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/nfs/operator.yaml#L124-L127

I think that is how the rook-ceph-operator deployment and code does it as well.

Reply:
As long as you're running inside the operator, you would have access to it. Another place in Rook does basically the same thing:
```
	// Getting the info of the operator pod
	pod, err := k8sutil.GetRunningPod(clientset)
	TerminateOnError(err, "failed to get pod")

	// Get the actual operator container image name
	containerImage, err := k8sutil.GetContainerImage(pod, containerName)
	TerminateOnError(err, "failed to get container image")
```

Reply:
Sorry folks, it took me a while to understand how to fix the test, and I still don't think I have it right. please take a look. Thanks